fbw-a32nx/src/systems/fmgc/src/components/FwcMessenger.ts

```diff
@@ -0,0 +1,34 @@
+import { getFlightPhaseManager } from '@fmgc/flightphase';
+import { FlightPlanManager } from '@fmgc/wtsdk';
+import { FmgcFlightPhase } from '@shared/flightphase';
+import { FmgcComponent } from './FmgcComponent';
+
+/**
+ * This is a placeholder for the FMGC messaging for the FWC system via DMC simulation
+ */
+export class FwcMessenger implements FmgcComponent {
+    private flightPlanManager: FlightPlanManager;
+
+    private transmittedMDA: Boolean;
+
+    init(_baseInstrument: BaseInstrument, flightPlanManager: FlightPlanManager): void {
+        this.flightPlanManager = flightPlanManager;
+    }
+
+    update(_deltaTime: number): void {
+        const flightPhase = getFlightPhaseManager().phase;
+
+        const currentDistanceToDestination = this.flightPlanManager.getDistanceToDestination();
+
+        const setMDA = SimVar.GetSimVarValue('L:A32NX_FMC_MINIMUM_DESCENT_ALTITUDE', 'feet');
+
+        // Only expose the MDA after being in Cruise phase and within 200NM of destination
+        if (currentDistanceToDestination <= 200 && flightPhase === FmgcFlightPhase.Cruise) {
```

Comment:
Keep in mind that it's possible that the aircraft never reaches the cruise phase and transitions directly from the climb to the descent phase.

Reply:
Yearh it should be Cruise or later phase. So it can handle going straight to descent, approach, G/A etc.